instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/internal/DeclarativeConfigTest.java

```diff
@@ -17,8 +17,10 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
+@Disabled // does not work after declarative config changes
```

Comment:
does this mean declarative config won't work in upcoming agent release?

Reply:
I just wasn't immediately able to figure out how to fix this and wanted to push the rest of the changes out to see whether something else still fails. I'll try to fix it but we may need to ask help from @jack-berg for this.

Reply:
Fixed the test, idk if I did  it correctly. Looking at https://github.com/open-telemetry/opentelemetry-configuration/blob/main/examples/kitchen-sink.yaml it seems that we need to group some of the resource providers. @jack-berg could you review.